File: day12/wordCount.cpp
#include <iostream>
#include <sstream>
#include <string>

int main() {
    std::string input;
    
    while (true) {
        std::cout << "Please type a sentence (one or more words): ";
        std::getline(std::cin, input);

        // Trim leading/trailing whitespace (optional step depending on input expectations)
        if (input.empty()) {
            std::cout << "Input cannot be empty.\n" << std::endl;
            continue;
        }

        std::istringstream stream(input);
        std::string word;
        int count = 0;

        while (stream >> word) {
            count++;
        }

        if (count == 1) {
            std::cout << "There is 1 word in your sentence." << std::endl;
        } else {
            std::cout << "There are " << count << " words in your sentence." << std::endl;
        }

        break;
    }

    return 0;
}

File: day12/word_count.rs
use std::io::{self, Write};

fn main() {
    loop {
        print!("Please type a sentence (one or more words): ");
        io::stdout().flush().unwrap(); // Make sure prompt prints before input

        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();

        let words: Vec<&str> = input
            .split_whitespace()
            .collect();

        let count = words.len();

        if count == 0 {
            println!("Input cannot be empty.\n");
            continue;
        } else if count == 1 {
            println!("There is 1 word in your sentence.");
        } else {
            println!("There are {} words in your sentence.", count);
        }

        break;
    }
}
